db/kv_checksum.h

```diff
@@ -0,0 +1,70 @@
+//  Copyright (c) 2020-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include "db/dbformat.h"
+#include "rocksdb/types.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+// Holds checksums of user keys/values and copies of internal metadata for
+// protecting integrity of data as it is processed.
```

Comment:
Can you explain somewhere why we need those metadata along the operation? It's a lot of information to keep in the class per KV.

Reply:
It's needed to verify the metadata wasn't corrupted. To save memory, we can fold all the checksums and metadata copies using XOR into a configurable number of bytes. As mentioned in the description, this PR only addresses `WriteBatch`, and does not implement such folding yet under the assumption that per-KV overhead is not important yet. Please let me know if you think that assumption is flawed (it is just my random guess).

Reply:
Done, we are now only storing the XOR of hashes.